packages/xo-web/src/xo-app/vm/tab-advanced.js

```diff
@@ -122,6 +128,33 @@ class ResourceSetItem extends Component {
   }
 }
 
+@addSubscriptions({
+  resourceSets: subscribeResourceSets,
+})
+class ShareVmButton extends Component {
+  _getResourceSet = createSelector(
+    () => this.props.resourceSets,
+    () => this.props.vm.resourceSet,
+    (resourceSets, id) =>
+      assign(find(resourceSets, { id }), { type: 'resourceSet' })
+  )
+
+  render () {
+    return (
+      <TabButton
+        btnStyle='primary'
+        handler={shareVm}
+        handlerParam={{
```

Comment:
The purpose of `handlerParam` is to be able to pass data easily without having to create new values (ie functions) in `render`.

Because you are creating a new object, you are not taking advantage of this, I suggest to use a proxy handler:
```js
_shareVm = () => {
  const { resourceSets, vm) = this.props
  return shareVm(vm, { ...find(resourceSets, { id: vm.resourceSet }), type: 'resourceSet' })
}
```